Cython/Compiler/Symtab.py

```diff
@@ -270,6 +270,11 @@ def __lt__(left, right):
     def cf_is_reassigned(self):
         return len(self.cf_assignments) > 1
 
+    def make_cpp_optional(self):
+        self.is_cpp_optional = True
```

Comment:
Seems worth another assertion here.
```suggestion
    def make_cpp_optional(self):
        assert self.type.is_cpp_class
        self.is_cpp_optional = True
```